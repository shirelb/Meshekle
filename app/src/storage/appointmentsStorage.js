import axios from "axios";
import {APP_SOCKET, SERVER_URL} from "../shared/constants";


var getUserAppointmentRequests = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/appointmentRequests/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {
                status: 'requested'
            },
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointment requests error ', error)
        });
};

var getUserAppointments = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/appointments/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {
                status: 'set'
            },
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointments error ', error)
        });
};

var getUserAppointmentById = function (userId, userHeaders, eventId) {
    return axios.get(`${SERVER_URL}/api/appointments/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {status: 'set', appointmentId: eventId}
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointments by id error ', error)
        });
};

var postUserAppointmentRequest = function (userId, serviceProvider, appointmentRequest, userHeaders) {
    return axios.post(`${SERVER_URL}/api/appointmentRequests/user/request`,
        {
            userId: userId,
            serviceProviderId: serviceProvider.serviceProviderId,
            role: serviceProvider.role,
            availableTime: appointmentRequest.availableTime,
            notes: appointmentRequest.notes ? appointmentRequest.notes : '',
            subject: JSON.stringify(appointmentRequest.subject)
        },
        {
            headers: userHeaders,
        })
        .then(response => {
            APP_SOCKET.emit('userPostAppointmentRequests', {
                serviceProviderId: serviceProvider.serviceProviderId
            });

            return response;
        })
        .catch(error => {
            console.log('post user appointment request error ', error)
        });
};

var cancelAppointmentRequestById = (appointmentRequest, headers) => {
    return axios.put(`${SERVER_URL}/api/appointmentRequests/user/reject`,
        {
            userId: appointmentRequest.AppointmentDetail.clientId,
            appointmentRequestId: appointmentRequest.requestId,
        },
        {
            headers: headers,
            params: {
                status: 'rejected'
            },
        }
    )
        .then((response) => {
            APP_SOCKET.emit('userCancelAppointmentRequests', {
                serviceProviderId: appointmentRequest.AppointmentDetail.serviceProviderId,
            });

            return response
        })
        .catch((error) => {
            console.log('reject appointment request error ', error);
        });
};

var cancelAppointmentById = (appointment, headers) => {
    return axios.put(`${SERVER_URL}/api/appointments/user/cancel/userId/${appointment.AppointmentDetail.clientId}/appointmentId/${appointment.appointmentId}`,
        {},
        {
            headers: headers
        }
    )
        .then((response) => {
            APP_SOCKET.emit('userCancelAppointment', {
                serviceProviderId: appointment.AppointmentDetail.serviceProviderId,
            });

            return response
        })
        .catch((error) => {
            console.log('cancel appointment error ', error);
        });
};


export default {
    getUserAppointments,
    getUserAppointmentById,
    postUserAppointmentRequest,
    getUserAppointmentRequests,
    cancelAppointmentRequestById,
    cancelAppointmentById,
};
