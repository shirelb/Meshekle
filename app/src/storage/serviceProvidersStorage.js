import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getServiceProviders = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('load appointments error ', error)
        });
};


export default {getServiceProviders};
