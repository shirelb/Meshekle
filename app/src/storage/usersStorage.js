import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getUserEvents = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/events/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user events error ', error)
        });
};

var getUserById = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var getUsers = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: userHeaders}
    )
        .then(response => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var userValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/users/validToken`,
        {
            "token": token,
        },
    );
}

var userLogin = function (userId, password) {
    return axios.post(`${SERVER_URL}/api/users/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
}


export default {getUserEvents, getUserById, userValidToken, userLogin, getUsers};
