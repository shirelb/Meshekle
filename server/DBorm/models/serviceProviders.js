module.exports = (sequelize, type) => {
    return sequelize.define('ServiceProviders', {
        serviceProviderId:{
            type: type.INTEGER,
            primaryKey: true,
            autoIncrement: true,
            allowNull: false,
            required: true
        },
        role : {
            type: type.STRING,
            allowNull: false,
            required: true
        },
        userId: {
            type: type.INTEGER,
            allowNull: false,
            required: true
        },
        operationTime: {
            type: type.STRING,
            allowNull: false,
            required: true
        },
        phoneNumber: {
            type: type.STRING,
            allowNull: false,
            required: true
        },
    });
};