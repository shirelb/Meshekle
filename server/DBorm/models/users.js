module.exports = (sequelize, type) => {
    return sequelize.define('Users', {
        userId:{
            type: type.STRING,
            primaryKey: true,
            required: true
        },
        fullname: {
            type: type.STRING,
            required: true
        },
        password: {
            type: type.STRING,
            required: true
        },
        email: {
            type: type.STRING,
            required: true
        },
        mailbox: {
            type: type.INTEGER,
            required: true
        },
        cellphone: {
            type: type.STRING,
            required: true
        },
        phone: {
            type: type.STRING
        },
        bornDate: {
            type: type.DATE,
            required: true
        },
        active: {
            type: type.BOOLEAN,
            required: true,
            defaultValue: true
        },
        image: {
            type: type.BLOB,
        }
    });
};