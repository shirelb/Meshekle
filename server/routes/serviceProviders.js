var express = require('express');
var router = express.Router();
const Sequelize = require('sequelize');
const Op = Sequelize.Op;
const {Users, ServiceProviders, AppointmentRequests, Appointments, Incidents, Events} = require('../DBorm/DBorm');

/* GET users listing. */
router.get('/', function(req, res, next) {
  res.send('respond with a resource serviceProviders');
});


/* POST user . */
router.post('/users/add', function (req, res, next) {
    Users.create({
        userId: req.body.userId,
        fullname: req.body.fullname,
        password: req.body.password,
        email: req.body.email,
        mailbox: req.body.mailbox,
        cellphone: req.body.cellphone,
        phone: req.body.phone,
        bornDate: new Date(req.body.bornDate),
        active: true,
    })
        .then((newUser) => {
            console.log('New user' + newUser.username + ', with id ' + newUser.userId + ' has been created.');
            res.status(200).send({"message": "User successfully added!", newUser});
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/* POST user . */
router.post('/add', function (req, res, next) {
    ServiceProviders.create({
        serviceProviderId: req.body.serviceProviderId,
        userId: req.body.userId,
        role: req.body.role,
        operationTime: req.body.operationTime,
        phoneNumber: req.body.phoneNumber,
        appointmentWayType: req.body.appointmentWayType,
        active: true,
    })
        .then((newUser) => {
            console.log('New user' + newUser.username + ', with id ' + newUser.userId + ' has been created.');
            res.status(200).send({"message": "User successfully added!", newUser});
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

module.exports = router;
