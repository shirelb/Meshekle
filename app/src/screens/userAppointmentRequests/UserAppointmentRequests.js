import React, {Component} from 'react';
import {Button, ScrollView, StyleSheet, Text} from 'react-native';
import {List} from "react-native-paper";
import phoneStorage from "react-native-simple-store";
import appointmentsStorage from "../../storage/appointmentsStorage";
import serviceProvidersStorage from "../../storage/serviceProvidersStorage";
import AppointmentRequestInfo from "../../components/appointmentRequest/AppointmentRequestInfo";
import {APP_SOCKET} from "../../shared/constants";
import {SearchBar} from "react-native-elements";


export default class UserAppointmentRequests extends Component {
    constructor(props) {
        super(props);

        this.state = {
            userAppointmentRequests: [],
            appointmentRequestSelected: {},
            formModal: false,
            infoModal: false,
            appointmentRequestDetails: {},
            noAppointmentRequestsFound: false,
        };

    }

    componentDidMount() {
        phoneStorage.get('userData')
            .then(userData => {
                // console.log('agenda componentDidMount userData ', userData);
                this.userHeaders = {
                    'Authorization': 'Bearer ' + userData.token
                };
                this.userId = userData.userId;
                this.loadMyAppointmentsRequests();
            });

        APP_SOCKET.on("getUserAppointmentRequests", this.loadMyAppointmentsRequests.bind(this));
    }

    componentWillUnmount() {
        APP_SOCKET.off("getUserAppointmentRequests");
    }

/*    componentWillReceiveProps(nextProps, nextContext) {
        this.loadMyAppointmentsRequests();
    }*/

    loadMyAppointmentsRequests() {
        appointmentsStorage.getUserAppointmentRequests(this.userId, this.userHeaders)
            .then((response) => {
                let userAppointmentRequests = response.data;

                console.log('userAppointmentRequests ', userAppointmentRequests);

                if (userAppointmentRequests.length === 0) {
                    this.setState({
                        userAppointmentRequests: userAppointmentRequests,
                    });
                    this.userAppointmentRequests = userAppointmentRequests
                }
                else
                    userAppointmentRequests.forEach(appointmentRequest => {
                        serviceProvidersStorage.getServiceProviderUserDetails(appointmentRequest.AppointmentDetail.serviceProviderId, this.userHeaders)
                            .then(user => {
                                appointmentRequest.serviceProviderFullname = user.data.fullname;
                                appointmentRequest.expanded = false;

                                this.setState({
                                    userAppointmentRequests: userAppointmentRequests,
                                });

                                this.userAppointmentRequests = userAppointmentRequests
                            })
                    })
            })
            .catch(err => console.log("loadMyAppointmentsRequests error ", err))
    };

    renderSeparator = () => {
        return (
            <View
                style={{
                    height: 1,
                    width: "86%",
                    backgroundColor: "#CED0CE",
                    marginLeft: "14%"
                }}
            />
        );
    };

    updateSearch = search => {
        console.log("in search ", search);
        // this.setState({search});
        let searchText = search.toLowerCase();
        let userAppointmentRequests = this.state.userAppointmentRequests;
        let filteredByNameOrRole = userAppointmentRequests.filter((item) => {
            return item.serviceProviderFullname.toLowerCase().match(searchText) || item.AppointmentDetail.role.toLowerCase().match(searchText);
        });
        // let filteredByRole = userAppointmentRequests.filter((item) => {
        //     return item.role.toLowerCase().match(searchText)
        // });
        if (!searchText || searchText === '') {
            this.setState({
                userAppointmentRequests: this.userAppointmentRequests
            })
        } else if (!Array.isArray(filteredByNameOrRole) && !filteredByNameOrRole.length) {
            // set no data flag to true so as to render flatlist conditionally
            this.setState({
                noAppointmentRequestsFound: true
            })
        } else if (Array.isArray(filteredByNameOrRole)) {
            this.setState({
                noAppointmentRequestsFound: false,
                userAppointmentRequests: filteredByNameOrRole
            })
        }
    };


    render() {
        // console.log('props ',this.props);

        return (
            <ScrollView>
                <Button
                    title="בקש תור חדש"
                    onPress={() => {
                        this.props.navigation.state.params.onAppointmentRequestPress();
                    }}
                />

                <SearchBar
                    placeholder="חפש..."
                    lightTheme
                    onChangeText={this.updateSearch.bind(this)}
                    // round
                />

                {/*{this.state.noAppointmentRequestsFound ?*/}
                {this.state.userAppointmentRequests.length === 0 ?
                    <Text>אין לך בקשות תורים</Text>
                    :
                    <List.Section>
                        {Array.isArray(this.state.userAppointmentRequests) &&
                        this.state.userAppointmentRequests.map((item) => {
                            return <List.Accordion
                                key={item.requestId}
                                // title={moment(item.startDateAndTime).format('HH:mm') + '-' + moment(item.endDateAndTime).format('HH:mm')}
                                title={item.AppointmentDetail.role}
                                description={item.serviceProviderFullname}
                                // left={props => <List.Icon {...props} icon="perm-contact-calendar"/>}
                                expanded={item.expanded}
                                onPress={() => {
                                    // let expanded = this.state.expanded;
                                    // expanded[item.date] = !expanded[item.date];
                                    // this.setState({expanded: expanded})
                                    let appointmentRequestSelected = item;
                                    appointmentRequestSelected.expanded = !appointmentRequestSelected.expanded;
                                    this.setState({
                                        appointmentRequestSelected: appointmentRequestSelected,
                                        appointmentRequestDetails: {}
                                    })
                                }}
                            >
                                <List.Item
                                    // key={item.requestId+'0'}
                                    title={"סטאטוס: " + item.status}
                                    // description={item.notes + ' \n ' + item.optionalTimes.toString()}
                                    containerStyle={{borderBottomWidth: 0}}
                                    // onPress={() => console.log("item was presssed!!  ", item)}
                                />
                                <List.Item
                                    // key={item.requestId+'1'}
                                    title={"נושא: " + JSON.parse(item.AppointmentDetail.subject).join(", ")}
                                    // description={item.notes + ' \n ' + item.optionalTimes.toString()}
                                    description={"לפרטים נוספים הקש כאן"}
                                    containerStyle={{borderBottomWidth: 0}}
                                    onPress={() => {
                                        this.setState({appointmentRequestDetails: item, infoModal: true})
                                    }}
                                />
                                {/*<List.Item
                                // key={item.requestId+'2'}
                                title={item.notes === "" ? "אין הערות" : "הערות:" + item.notes}
                                // description={item.notes + ' \n ' + item.optionalTimes.toString()}
                                containerStyle={{borderBottomWidth: 0}}
                                onPress={() => console.log("item was presssed!!  ", item)}
                            />
                            <List.Item
                                // key={item.requestId+'3'}
                                title={JSON.parse(item.optionalTimes).map(function (elem) {
                                    return elem.date + ": " + elem.hours.map(function (h) {
                                        return h.startHour + '-' + h.endHour
                                    }).join(', ');
                                }).join("; ")}
                                // description={item.notes + ' \n ' + }
                                containerStyle={{borderBottomWidth: 0}}
                                onPress={() => console.log("item was presssed!!  ", JSON.parse(item.optionalTimes).join(": "))}
                            />*/}
                                {/*<List.Item
                                // key={item.requestId+'3'}
                                // title={"לפרטים נוספים"}
                                description={"לפרטים נוספים הקש כאן"}
                                containerStyle={{borderBottomWidth: 0}}
                                onPress={() => {
                                    this.setState({appointmentRequestSelected: item, infoModal: true})
                                }}
                            />*/}
                            </List.Accordion>
                        })
                        }
                    </List.Section>
                }

                {this.state.appointmentRequestDetails.requestId ?
                    <AppointmentRequestInfo
                        appointmentRequest={this.state.appointmentRequestDetails}
                        modalVisible={this.state.infoModal}
                    />
                    : null
                }
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    }
});

