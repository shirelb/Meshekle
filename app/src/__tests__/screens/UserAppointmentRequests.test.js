import React from 'react';

import {FlatList} from 'react-native';
import {List} from "react-native-paper";

import {shallow} from "enzyme/build";
import serviceProvidersStorage from "../../storage/serviceProvidersStorage";
import usersStorage from "../../storage/usersStorage";
import user013637605AppointmentRequests from "../jsons/user013637605AppointmentRequests";
import users from "../jsons/users";
import serviceProviders from "../jsons/serviceProviders";
import UserAppointmentRequests from "../../screens/userAppointmentRequests/UserAppointmentRequests";
import phoneStorage from "react-native-simple-store";
import appointmentsStorage from "../../storage/appointmentsStorage";


jest.mock("react-native-simple-store");
jest.mock("../../storage/usersStorage");
jest.mock("../../storage/serviceProvidersStorage");
jest.mock("../../storage/appointmentsStorage");


describe("UserAppointmentRequests should", () => {
    let wrapper = null;
    let componentInstance = null;
    const props = {};
    const userTest = users[698];
    const mockStore = {
        userData: {
            userId: "549963652",
            token: "some token"
        }
    };

    phoneStorage.get = jest.fn().mockImplementation((key) => Promise.resolve(mockStore[key]));
    usersStorage.getUsers = jest.fn().mockImplementation(() => Promise.resolve(users));
    usersStorage.getUserByUserID = jest.fn().mockImplementation((userId) => Promise.resolve(users.filter(user => user.userId === userId)[0]));
    serviceProvidersStorage.getServiceProviders = jest.fn().mockResolvedValue(serviceProviders);
    serviceProvidersStorage.getServiceProviderUserDetails = jest.fn().mockImplementation((serviceProviderId) => Promise.resolve(
        {data: users.filter(user => user.userId === serviceProviderId)[0]}));
    serviceProvidersStorage.getServiceProviderById = jest.fn().mockImplementation((serviceProviderId) => Promise.resolve(serviceProviders.filter(provider =>
        provider.serviceProviderId === serviceProviderId)));
    appointmentsStorage.getUserAppointmentRequests = jest.fn().mockResolvedValue({data: user013637605AppointmentRequests});
    appointmentsStorage.cancelAppointmentRequestById = jest.fn().mockResolvedValue();

    let userRequestsTest = user013637605AppointmentRequests;
    userRequestsTest[0].serviceProviderFullname = "Tara";
    userRequestsTest[1].serviceProviderFullname = "Gil";

    beforeAll(async (done) => {
        wrapper = await shallow(<UserAppointmentRequests/>);
        componentInstance = wrapper.instance();

        await wrapper.update();
        done();
    });

    afterAll(() => {
    });

    beforeEach(async (done) => {
        jest.clearAllMocks();
        await componentInstance.setState({userAppointmentRequests: userRequestsTest});
        await wrapper.update();
        done();
    });

    afterEach( async (done) => {
        await componentInstance.setState({userAppointmentRequests: userRequestsTest});
        await wrapper.update();
        done();
    });

    it('match snapshot', async () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("mounted with the right data", () => {

        expect(componentInstance.state.userAppointmentRequests.length).toEqual(2);
        expect(componentInstance.state.formModal).toEqual(false);
        expect(componentInstance.state.infoModal).toEqual(false);
        expect(componentInstance.state.appointmentRequestSelected).toEqual({});
        expect(componentInstance.state.appointmentRequestDetails).toEqual({});
        expect(componentInstance.state.noAppointmentRequestsFound).toEqual(false);
    });

    it("render with what the user see", async () => {

        expect(wrapper.find('Button')).toHaveLength(1);
        expect(wrapper.find('Button').props().title).toEqual('בקש תור חדש');
        expect(wrapper.find('Text')).toHaveLength(0);
        expect(wrapper.find(List.Section)).toHaveLength(1);
        expect(wrapper.find(List.Accordion)).toHaveLength(2);
        expect(wrapper.find(List.Item)).toHaveLength(4);
        expect(wrapper.find('AppointmentRequestInfo')).toHaveLength(0);
    });

    it("search for appointment requests with service provider with name that has ra", async () => {
        const updateSearchSpy = jest.spyOn(componentInstance, 'updateSearch');

        await componentInstance.updateSearch('ra');

        expect(updateSearchSpy).toHaveBeenCalled();
        expect(componentInstance.state.userAppointmentRequests.length).toEqual(1);
    });

    it("cancel Appointment Request on click erase icon", async () => {
        wrapper = shallow(<UserAppointmentRequests/>);
        componentInstance = wrapper.instance();

        const cancelAppointmentRequestSpy = jest.spyOn(componentInstance, 'cancelAppointmentRequest');

        componentInstance.cancelAppointmentRequest(userRequestsTest[0]);

        expect(cancelAppointmentRequestSpy).toHaveBeenCalled();
        expect(cancelAppointmentRequestSpy).toHaveBeenCalledWith(userRequestsTest[0]);
    });

    it("close Appointment Request Info", async () => {
        wrapper = shallow(<UserAppointmentRequests/>);
        componentInstance = wrapper.instance();

        const closeAppointmentRequestInfoSpy = jest.spyOn(componentInstance, 'closeAppointmentRequestInfo');

        componentInstance.closeAppointmentRequestInfo(userTest);

        expect(closeAppointmentRequestInfoSpy).toHaveBeenCalled();
        expect(componentInstance.state.infoModal).toEqual(false);
        expect(componentInstance.state.appointmentRequestDetails).toEqual({});
    });

});
