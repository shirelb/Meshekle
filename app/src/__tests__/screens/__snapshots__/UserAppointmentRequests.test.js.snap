// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAppointmentRequests should match snapshot 1`] = `
<ScrollViewMock
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
>
  <Button
    onPress={[Function]}
    title="בקש תור חדש"
  />
  <Search
    icon={Object {}}
    lightTheme={true}
    loadingIcon={Object {}}
    noIcon={false}
    onChangeText={[Function]}
    placeholder="חפש..."
    placeholderTextColor="#86939e"
    round={false}
    showLoadingIcon={false}
  />
  <withTheme(List.Section)>
    <View
      key="907"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <Icon
          color="red"
          name="delete-forever"
          onPress={[Function]}
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      </View>
      <View
        style={
          Object {
            "width": "70%",
          }
        }
      >
        <withTheme(List.Accordion)
          description="Tara"
          onPress={[Function]}
          title="מספרה"
        >
          <withTheme(List.Item)
            containerStyle={
              Object {
                "borderBottomWidth": 0,
              }
            }
            title="סטאטוס: נדחה"
          />
          <withTheme(List.Item)
            containerStyle={
              Object {
                "borderBottomWidth": 0,
              }
            }
            description="לפרטים נוספים הקש כאן"
            onPress={[Function]}
            title="נושא: פן, תספורת, צבע, גוונים"
          />
        </withTheme(List.Accordion)>
      </View>
    </View>
    <View
      key="961"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <Icon
          color="red"
          name="delete-forever"
          onPress={[Function]}
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      </View>
      <View
        style={
          Object {
            "width": "70%",
          }
        }
      >
        <withTheme(List.Accordion)
          description="Gil"
          onPress={[Function]}
          title="מרפאת שיניים"
        >
          <withTheme(List.Item)
            containerStyle={
              Object {
                "borderBottomWidth": 0,
              }
            }
            title="סטאטוס: אושר"
          />
          <withTheme(List.Item)
            containerStyle={
              Object {
                "borderBottomWidth": 0,
              }
            }
            description="לפרטים נוספים הקש כאן"
            onPress={[Function]}
            title="נושא: טיפול שורש, עקירה, שיננית, יישור שיניים"
          />
        </withTheme(List.Accordion)>
      </View>
    </View>
  </withTheme(List.Section)>
</ScrollViewMock>
`;
